--- Dh_Lib.pm.orig	2018-05-10 11:11:57.000000000 +0200
+++ Dh_Lib.pm	2020-03-27 17:10:17.000000000 +0100
@@ -31,7 +35,7 @@
 	'DBGSYM_PACKAGE_TYPE' => 'ddeb',
 };
 
-use Errno qw(ENOENT);
+use Errno qw(ENOENT EXDEV);
 use Exporter qw(import);
 use File::Glob qw(bsd_glob GLOB_CSH GLOB_NOMAGIC GLOB_TILDE);
 our (@EXPORT, %dh);
@@ -535,8 +650,16 @@
 	}
 	return 1 if $dh{NO_ACT};
 	if (not rename($source, $dest)) {
-		my $files = escape_shell($source, $dest);
-		error("mv $files: $!")
+		my $ok = 0;
+		if ($! == EXDEV) {
+			# Replay with a fork+exec to handle crossing two mount
+			# points (See #897569)
+			$ok = _doit('mv', $source, $dest);
+		}
+		if (not $ok) {
+			my $files = escape_shell($source, $dest);
+			error("mv $files: $!");
+		}
 	}
 	return 1;
 }
